python/python/tests/test_vector_index.py

```diff
@@ -185,8 +187,42 @@ def test_pre_populated_ivf_centroids(dataset, tmp_path: Path):
     )
 
     q = np.random.randn(128)
-    actual = dataset.to_table(
+    actual = dataset_with_index.to_table(
         columns=["id"],
         nearest={"column": "vector", "q": q, "k": 10, "use_index": False},
     )["id"].to_numpy()
     assert len(actual) == 10
+
+    index_uuid = dataset_with_index.list_indices()[0]["uuid"]
+    assert len(index_uuid) == 36
+
+    expected_filepath = str(tmp_path / "_indices" / index_uuid / "index.idx")
+    if platform.system() == "Windows":
+        expected_filepath = expected_filepath.replace("\\", "/")
+    expected_index = [
+        {
+            "index_type": "Ivf",
+            "uuid": index_uuid,
+            "uri": expected_filepath,
+            "metric_type": "l2",
+            "num_partitions": 5,
+            "sub_index": '{"index_type":"PQ","nbits":8,"num_sub_vectors":8,'
+            '"dimension":128,"metric_type":"l2"}',
+        }
+    ]
+
+    assert dataset_with_index.statistics("non-existent_idx") == "[\n\n]"
+    assert dataset_with_index.statistics("") == "[\n\n]"
+
+    actual_index = json.loads(dataset_with_index.statistics())
+    actual_index[0].pop("partitions")
+    assert actual_index == expected_index
+
+    actual_index = json.loads(dataset_with_index.statistics("vector_idx"))
```

Comment:
In Python, could we do the `json.loads()` step for the user?

Reply:
Changed.